drivers/dma/dma_stm32u5.c

```diff
@@ -563,6 +563,32 @@ static int dma_stm32_start(const struct device *dev, uint32_t id)
 	return 0;
 }
 
+static int dma_stm32_suspend(const struct device *dev, uint32_t id)
+{
+	const struct dma_stm32_config *config = dev->config;
+	DMA_TypeDef *dma = (DMA_TypeDef *)(config->base);
+
+	/* Give channel from index 0 */
+	id = id - STM32_DMA_STREAM_OFFSET;
```

Comment:
Nitpick: Since STM32_DMA_STREAM_OFFSET is 0 for U5 and driverr is IP specific ( not mix with other IP), I think it would enhence readability to clean up usage of STM32_DMA_STREAM_OFFSET in that specific file.
(Could be done in another PR)